lib/msfl_visitors/visitors/base.rb

```diff
@@ -1,15 +1,19 @@
 module MSFLVisitors
   module Visitors
     class Base
-      attr_accessor :collector
 
-      def visit(obj)
-        send("visit_#{obj.class.to_s.gsub('::', '_')}", obj, collector)
+      def initialize(collector, renderer)
+        @collector = collector
+        @renderer = renderer
       end
 
-      def initialize(collector)
-        self.collector = collector
+      def visit(obj)
+        collector << renderer.render(obj)
       end
+
+      private
+
+      attr_reader :collector, :renderer
```

Comment:
I prefer to set instance vars in the constructor, and then use private accessors everywhere. I've run into a lot of cases where I wind up replacing the attr_reader with a bigger method, and this means I don't have to search/replace.